site/frontend/src/pages/compare/compile/common.ts

```diff
@@ -160,7 +160,7 @@ export function computeCompileComparisonsWithNonRelevant(
   function shouldShowTestCase(comparison: TestCaseComparison<CompileTestCase>) {
     const name = `${comparison.testCase.benchmark} ${comparison.testCase.profile} ${comparison.testCase.scenario} ${comparison.testCase.backend}`;
     const nameFilter = filter.name && filter.name.trim();
-    const nameFiltered = !nameFilter || name.includes(nameFilter);
+    const nameFiltered = !nameFilter || name.match(new RegExp(nameFilter));
```

Comment:
When the input is not a valid regex, we should somehow catch the error, and treat the input as a normal string:
![image](https://github.com/rust-lang/rustc-perf/assets/4539057/390ee64c-4f9c-4c39-8db3-4f43186dc2bd)

Because when this happens, it sadly breaks the page.

Also, it would be nice to treat the types a bit more explicitly here, i.e. compare that the result of match `!== null`, rather than implicitly coercing an array to a bool.